### ISSUE TITLE
kube-dns pod fails to start: Failed to create a kubernetes client: invalid configuration: no configuration has been provided

### ISSUE THREAD

**Issue by dfcowell** [kubernetes/minikube#1430]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

**Minikube version** (use `minikube version`): v0.18.0

**Environment**:
- **OS** Windows 10 Pro Version 1703 (Build 15063.138)
- **VM Driver**: Hyper-V
- **ISO version**: v0.18.0
- **Install tools**:
- **Others**:

**What happened**:

- Ran `minikube start --vm-driver=hyperv --hyperv-virtual-switch=minikube`
- Enabled the Ingress feature
- Deployed a set of containers, services and an ingress
- One container needs to connect to the internet as part of its startup process.
- Log filled with getaddrinfo ENOTFOUND errors
- Further investigation revealed that the `kube-dns-v20` RC was in an error state.

Logs from the failing pod:

**kubedns**
```
2017-05-01T03:31:41.6273247Z I0501 03:31:41.625567       1 dns.go:42] version: v1.6.0-alpha.0.680+3872cb93abf948-dirty
2017-05-01T03:31:41.6273884Z F0501 03:31:41.626877       1 server.go:54] Failed to create a kubernetes client: invalid configuration: no configuration has been provided
```

**dnsmasq**
```
2017-05-01T03:30:11.5547107Z dnsmasq[1]: started, version 2.76 cachesize 1000
2017-05-01T03:30:11.5547951Z dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth no-DNSSEC loop-detect inotify
2017-05-01T03:30:11.5548038Z dnsmasq[1]: using nameserver 127.0.0.1#10053
2017-05-01T03:30:11.554808Z dnsmasq[1]: read /etc/hosts - 7 addresses
```

**healthz**
```
2017-05-01T03:31:11.0407718Z 2017/05/01 03:31:11 Healthz probe on /healthz-dnsmasq error: Result of last exec: nslookup: can't resolve 'kubernetes.default.svc.cluster.local'
2017-05-01T03:31:11.0408243Z , at 2017-05-01 03:30:39.0109608 +0000 UTC, error exit status 1
```

**What you expected to happen**:

I expected the kube-dns pod to come up successfully and allow internal/external DNS resolution to work.

**How to reproduce it** (as minimally and precisely as possible):

See above.

**Anything else do we need to know**:

Pods can reach the internet (e.g. `ping 8.8.8.8` works reliably).
Everything else on the cluster appears to work.

**Comment by dfcowell** [comment]
There seems to be a more widespread issue in the cluster - after bringing it back up to work on it again (I shut it down with `minikube stop`) the dashboard is no longer starting successfully. It was working previously.

The error in question is:

```
Error while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
```

Sure enough, these env vars don't exist in any running containers. Not sure where exactly to go looking now - I'm not familiar enough with the Kubernetes internals to dig deeply into this issue.

**Comment by aaron-prindle** [comment]
Can you post the output of `minikube logs`?

**Comment by dfcowell** [comment]
I've deleted and re-created the VM a few times since I posted the issue, so I don't have access to the `minikube logs` output from when the issue occurred right now. It happens seemingly randomly, though often enough that I'm sure I'll run into it again in the next few days. A coworker ran into the same issue on his OSX setup, so I might be able to grab the logs from him.

It seems like a race condition might be causing the issue, but once it happens the cluster is unrecoverable and only deleting and recreating the minikube instance can resolve the problem. Restarting has no effect.

Sorry for the lack of logs - I'll post them as soon as I run into the problem again.
